Libraries/LibWeb/Animations/Animatable.cpp

```diff
@@ -187,14 +187,14 @@ void Animatable::visit_edges(JS::Cell::Visitor& visitor)
 
 GC::Ptr<CSS::CSSStyleDeclaration const> Animatable::cached_animation_name_source(Optional<CSS::Selector::PseudoElement::Type> pseudo_element) const
 {
-    if (pseudo_element.has_value())
+    if (pseudo_element.has_value() && CSS::Selector::PseudoElement::is_known_pseudo_element_type(pseudo_element.value()))
```

Comment:
This change means that requesting the animation name source for a fake pseudo-elements will give you its element's one instead, which is wrong - it should give you nothing.

The same for the other places you've added this check. It's not correct to fall back to returning the element's data.

Reply:
That makes sense, that is not as easy, as there arise a few question on how to do that in the best way:
- The setters `set_cached_animation_name_animation` and `set_cached_animation_name_source`  need to be checked too, as far as I understand, or am I wrong?
- the getters return a `GC::Ptr`, is it okay to return `{}` (alias empty Ptr alias "nullptr")

I did this in a separate commit, so that it can be seen, what the changes were, I'll squash it, when it's correct 😄